### ISSUE TITLE
WithConstructorArgument with invalid name still resolved instance

### ISSUE THREAD

**Issue by slaneyrw** [ninject/Ninject#226]
One of our devs accidentally configured a binding using WithConstructorArgument with a parameter name that didn't existing on the constructor.

Ninject happily built the object without exception.  Is this expected behaviour or should have it thrown an exception due to a mis-config ?

**Comment by BrunoJuchli** [comment]
It's the expected behavior - has been like this for years. I don't think it's ideal but then again when you have Parameters with `IsInherited == true` then there's not really a way around this behavior.

**Comment by slaneyrw** [comment]
thanks for the confirmation.  Not sure inheritance plays a part with constructor args though, you can't directly use the base class constructor.

**Comment by BrunoJuchli** [comment]
My argument is not related to class inheritance. [See the `IParameter` interface](https://github.com/ninject/Ninject/blob/master/src/Nfinject/Parameters/IParameter.cs). A constructor argument is just one of the `IParameter` implementations.

**Comment by StingyJack** [comment]
An explicit instruction was given to use a named ctor arg of a specific type. 

I think it would be expected that ninject would throw an error if it doesnt find a matching constructor - or at least configurable to do so.

I don't want to find out later in the application lifetime when it starts experiencing weird side effects (like I did yesterday and today, only to realize its a minor naming difference for something that has no other constructors).

**Comment by BrunoJuchli** [comment]
@StingyJack makes sense. However, as mentioned before with the current design there's no easy fix for this.
However, there's something you as a user can do.

- slightest change, improvement is to use a [TypeMatchingConstructorArgument](https://github.com/ninject/Ninject/blob/master/src/Ninject/Parameters/TypeMatchingConstructorArgument.cs) instead of a name matching one. This is possible as long as you don't have multiple arguments with the same type. 
- bigger change but full control is to use `ToMethod` binding to create the type
- possibly you could also use the `ToConstructor` binding.
